fuel-merkle/src/binary/in_memory.rs

```diff
@@ -62,6 +64,80 @@ impl Default for MerkleTree {
     }
 }
 
+#[derive(Default, Debug, Clone)]
+pub struct MerkleRootCalculator {
+    stack: Vec<Node>,
+    second_top_on_stack: Option<Node>,
+}
+
+impl MerkleRootCalculator {
+    pub fn new() -> Self {
+        Self {
+            stack: Vec::new(),
+            second_top_on_stack: None,
```

Comment:
Internally I feel that we should be able to do that without an additional `second_top_on_stack` field. The stack structure itself and knowing each height should be enough to know when we need to merge.

@bvrooman  Could you validate the algorithm, please? Maybe you can see how we can simplify it

Reply:
removed